src/app/selectors.js

```diff
@@ -372,37 +372,53 @@ const filterModelData = (filters, modelData, controllers) => {
     return clonedModelData;
   }
   const filterSegments = {};
-  // Collect segments
-  filters.forEach((filter) => {
-    const values = filter.split(":");
-    if (!filterSegments[values[0]]) {
-      filterSegments[values[0]] = [];
+
+  // Collect segments from filter data
+  Object.entries(filters).forEach((filter) => {
+    if (filter[1].length === 0) return;
+    if (!filterSegments[filter[0]]) {
+      filterSegments[filter[0]] = [];
     }
-    filterSegments[values[0]].push(values[1]);
+    filterSegments[filter[0]].push(filter[1]);
   });
 
   Object.entries(clonedModelData).forEach(([uuid, data]) => {
-    const remove = Object.entries(filterSegments).some(([segment, values]) => {
-      switch (segment) {
-        case "cloud":
-          return !values.includes(extractCloudName(data.model.cloudTag));
-        case "credential":
-          if (data.info) {
-            return !values.includes(
-              extractCredentialName(data.info.cloudCredentialTag)
-            );
-          }
-          break;
-        case "region":
-          return !values.includes(data.model.region);
-        case "owner":
-          if (data.info) {
-            return !values.includes(extractOwnerName(data.info.ownerTag));
-          }
-          break;
+    const modelName = data?.model?.name;
+    const cloud = data?.model && extractCloudName(data.model.cloudTag);
+    const credential =
+      data?.info && extractCredentialName(data.info.cloudCredentialTag);
+    const region = data?.model && data.model.region;
+    const owner = data?.info && extractOwnerName(data.info.ownerTag);
+    // Combine all of the above to create string for fuzzy custom search
+    const combinedModelAttributes = `${modelName} ${cloud} ${credential} ${region} ${owner}`;
+
+    const remove = Object.entries(filterSegments).some(
+      ([segment, valuesArr]) => {
+        const values = valuesArr[0];
+        switch (segment) {
+          case "cloud":
+            return !values.includes(cloud);
```

Comment:
Using `indexOf` is [dramatically faster](https://stackoverflow.com/questions/5296268/fastest-way-to-check-a-string-contain-another-substring-in-javascript) particularly in Chrome.

Reply:
Those are old benchmarks (and micro benchmarks at that). It is a very tiny bit slower now but not enough to make a difference as this isn't a hot path.